### DISCUSSION TITLE
How to verify signed transaction after signTx()?

### DISCUSSION THREAD

**Discussion by golddydev** [cardano-foundation/cardano-connect-with-wallet#88] [Q&A] [answered]
I am now building cardano dapp.
And I was trying to verify users by using `signData()` method.
But I found that the ledger type wallet (not seed-phrase) doesn't support the signData() method yet.

I was trying to solve this issue by replacing the `signData()` to `signTx()` method.

> I build the transaction who input is from zero address and output is signer's address (which seems to cost nothing for signers)

And then how can I verify this signed transaction data?

I used the same way I used for `signData()` but it is not working.

I used this module.
```
import verifyDataSignature from '@cardano-foundation/cardano-verify-datasignature';
```

But I got error like this.
```
Error: Public key not found
```

Thank you.

**Comment by fabianbormann** [comment]
@golddydev sorry I was not pinged about a new discussion from github 🙈 ... Did I understand correctly that you wanted to use a hardware wallet to sign data? This is currently not supported by the hardware wallets. SignTx is a different story and therefore it will not work with verify-datasignature.

**Reply by golddydev** [reply, marked as answer]
@fabianbormann , Thanks for you reply.
I implemented this function using signTx for ledger wallets.
And just use transaction related to zero address and do NOT submit that signed transaction, just use it for verification.
It all works fine, now.
